### PR TITLE
Add simple decorator for handling common botocore/boto3 errors

### PR DESCRIPTION
**Pull request by tremble**
##### SUMMARY

Adds a handler for common IAM API errors

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

plugins/module_utils/iam.py
plugins/modules/iam_user.py

##### ADDITIONAL INFORMATION

TODO:

- [x]  Changelog

### CODE REVIEW

File: plugins/module_utils/iam.py [hunk 1]
```diff
@@ -27,6 +28,68 @@ class AnsibleIAMError(AnsibleAWSError):
     pass
 
 
+def iam_common_error_handler(description):
+    """A simple error handler that catches the standard Boto3 exceptions and raises
```
Comment (alinabuzachis) [inline plugins/module_utils/iam.py:32]: I was only wondering if we could make it more generic and not only for IAM. I like the decorator's approach for error handling.

Comment (tremble) [reply plugins/module_utils/iam.py:32]: I was wondering the same thing, I didn't want to dig too deep if they wasn't a direction the wider team wanted things to go, but can certainly poke at it.

**Comment by tremble** [comment]
recheck

File: tests/unit/module_utils/errors/aws_error_handler/test_list_handler.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+
+import pytest
+
+from ansible_collections.amazon.aws.plugins.module_utils.botocore import HAS_BOTO3
+from ansible_collections.amazon.aws.plugins.module_utils.botocore import is_boto3_error_code
+from ansible_collections.amazon.aws.plugins.module_utils.errors import AWSErrorHandler
+from ansible_collections.amazon.aws.plugins.module_utils.exceptions import AnsibleAWSError
+
+if not HAS_BOTO3:
+    pytestmark = pytest.mark.skip("test_awsretry.py requires the python modules 'boto3' and 'botocore'")
```
Comment (hakbailey) [inline tests/unit/module_utils/errors/aws_error_handler/test_list_handler.py:19]:
```suggestion
    pytestmark = pytest.mark.skip("test_list_handler.py requires the python modules 'boto3' and 'botocore'")
```

File: plugins/module_utils/errors.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+
+            return handler
+
+        return wrapper
+
+    @classmethod
+    def list_error_handler(cls, description, default_value=None):
+        """A simple error handler that catches the standard Boto3 exceptions and raises
+        an AnsibleAWSError exception.
+        Error codes representing a non-existent entity will result in None being returned
```
Comment (hakbailey) [inline plugins/module_utils/errors.py:50]: Could we update this docstring to note that `default_value` can be passed to change the return value to, for example, an empty list? This is a clever feature that I think would be easy for developers to miss.

File: tests/unit/module_utils/iam/test_iam_error_handler.py [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+    pass
+
+import pytest
+
+from ansible_collections.amazon.aws.plugins.module_utils.botocore import HAS_BOTO3
+from ansible_collections.amazon.aws.plugins.module_utils.iam import AnsibleIAMError
+from ansible_collections.amazon.aws.plugins.module_utils.iam import IAMErrorHandler
+
+if not HAS_BOTO3:
+    pytestmark = pytest.mark.skip("test_awsretry.py requires the python modules 'boto3' and 'botocore'")
```
Comment (hakbailey) [inline tests/unit/module_utils/iam/test_iam_error_handler.py:18]:
```suggestion
    pytestmark = pytest.mark.skip("test_iam_error_handler.py requires the python modules 'boto3' and 'botocore'")
```

File: plugins/module_utils/errors.py (hunk 3 above)
Comment (tremble) [reply plugins/module_utils/errors.py:56]: done

**Review by hakbailey — APPROVED** [review]

**Review by alinabuzachis — APPROVED** [review]

**Pull request merged**